packages/ckeditor5-dev-release-tools/tests/tasks/preparerepository.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const expect = require( 'chai' ).expect;
+const sinon = require( 'sinon' );
+const mockery = require( 'mockery' );
+
+describe( 'dev-release-tools/tasks', () => {
+	describe( 'prepareRepository()', () => {
+		const pkgJsonTemplatePathPattern = /packages\/ckeditor5-dev-release-tools\/lib\/templates\/release-package\.json$/;
+
+		const pkgJsonTemplate = {
+			name: null,
+			version: null,
+			description: null,
+			keywords: null,
+			engines: null,
+			author: null,
+			license: null,
+			homepage: null,
+			bugs: null,
+			repository: null
+		};
+
+		const packages = [
+			'ckeditor5-core',
+			'ckeditor5-utils'
+		];
+
+		let options, stubs, prepareRepository;
+
+		beforeEach( () => {
+			options = {
+				pkgJsonContent: {
+					name: 'CKEditor5',
+					description: 'Foo bar baz.',
+					keywords: [ 'foo', 'bar', 'baz' ]
+				},
+				rootFilesToCopy: [
+					'src',
+					'CHANGELOG.md'
+				]
+			};
+
+			function stubReject( ...args ) {
+				if ( args.length === 0 ) {
+					throw new Error( 'Expected to receive an argument.' );
+				}
+
+				throw new Error( `No output configured for the following args: ${ args }` );
+			}
+
+			stubs = {
+				fs: {
+					ensureDir: sinon.stub().resolves(),
+					writeJson: sinon.stub().resolves(),
+					copy: sinon.stub().resolves(),
+
+					readJson: sinon.stub().callsFake( stubReject ),
+					readdir: sinon.stub().callsFake( stubReject )
+				},
+				process: {
+					cwd: sinon.stub( process, 'cwd' ).returns( 'current/working/dir' )
+				}
+			};
+
+			stubs.fs.readJson.withArgs( sinon.match( pkgJsonTemplatePathPattern ) ).resolves( pkgJsonTemplate );
+			stubs.fs.readdir.withArgs( 'current/working/dir/packages' ).resolves( packages );
```

Comment:
In such a case, I would assume it will be clearer if you use `callsFake()` instead of `withArgs()` and, depending on an argument, return a proper value.

Reply:
I would aim to remove regexps from tests and use `String.endsWith()` instead.